From 36d054bd0248de464fa9b80fc6f1528885d4f954 Mon Sep 17 00:00:00 2001
From: Sridharan S N <quic_sridsn@quicinc.com>
Date: Wed, 8 Nov 2023 14:27:25 +0530
Subject: [PATCH] mtd: nand: Add support for GD5F2GM7REYIG and MX35UF1G24AD
 device

Add support for gigadevice - GD5F2GM7REYIG and macronix
MX35UF1G24AD SPI NAND.

Change-Id: I3732242e87f7a9d71da5553249a2f21973e68193
Signed-off-by: Sridharan S N <quic_sridsn@quicinc.com>
---
 drivers/mtd/nand/raw/nand_ids.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/mtd/nand/raw/nand_ids.c b/drivers/mtd/nand/raw/nand_ids.c
index 432596e31da0..1b4fdbd87f70 100644
--- a/drivers/mtd/nand/raw/nand_ids.c
+++ b/drivers/mtd/nand/raw/nand_ids.c
@@ -93,6 +93,9 @@ struct nand_flash_dev nand_flash_ids[] = {
 	{"GD5F1GM7REYIG SPI NAND 1G-BIT",
 		{ .id = {0xc8, 0x81} },
 		SZ_2K, SZ_128, SZ_128K, 0, 2, 128, NAND_ECC_INFO(8, SZ_512)},
+	{"GD5F2GM7REYIG SPI NAND 2G-BIT",
+		{ .id = {0xc8, 0x82} },
+		SZ_2K, SZ_256, SZ_128K, 0, 2, 128, NAND_ECC_INFO(8, SZ_512)},
 	{"F50D1G41LB(2M) SPI NAND 1G 1.8V",
 		{ .id = {0xc8, 0x11} },
 		SZ_2K, SZ_128, SZ_128K, 0, 2, 64, NAND_ECC_INFO(4, SZ_512)},
@@ -131,6 +134,10 @@ struct nand_flash_dev nand_flash_ids[] = {
 		{ .id = {0xef, 0xba, 0x20} },
 		SZ_2K, SZ_64, SZ_128K, 0, 3, 64, NAND_ECC_INFO(4, SZ_512)},
 
+	{"MX35UF1G24AD SPI NAND 1G 1.8V",
+		{ .id = {0xc2, 0x94} },
+		SZ_2K, SZ_128, SZ_128K, 0, 2, 128, NAND_ECC_INFO(8, SZ_512)},
+
 	LEGACY_ID_NAND("NAND 4MiB 5V 8-bit",   0x6B, 4, SZ_8K, SP_OPTIONS),
 	LEGACY_ID_NAND("NAND 4MiB 3,3V 8-bit", 0xE3, 4, SZ_8K, SP_OPTIONS),
 	LEGACY_ID_NAND("NAND 4MiB 3,3V 8-bit", 0xE5, 4, SZ_8K, SP_OPTIONS),
-- 
2.34.1

